Update RELEASE.md (#1883)
kedro-org/kedro: RELEASE.md

## Bug fixes and other changes

> * Fixed `format` in `save_args` for `SparkHiveDataSet`, previously it does not allows you to save it as delta format.

Reviewer: ```suggestion
* Fixed `format` in `save_args` for `SparkHiveDataSet`, previously it didn't allow you to save it as delta format.
```

Author: Ops
